### PR TITLE
Deploy workflow and release instructions

### PR DESCRIPTION
**Pull request by nicoddemus**
Added a new `deploy` workflow, which publishes to PyPI when a new tag is pushed.

### CODE REVIEW

File: RELEASING.rst [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+    bumpversion minor --new-version 1.0.11
+
+   Changing ``minor`` and ``--new-version`` above accordingly.
+
+3. Push a branch with the changes.
+4. Once all builds pass and at least another maintainer approves, push a tag to ``upstream`` in the format ``v1.0.11`.
+   This will deploy to PyPI.
+5. Merge the PR (do not squash, to keep the tag).
+6. Create a new release on GitHub, posting the contents of the current CHANGELOG.
```
Comment (nicoddemus) [inline RELEASING.rst:16]: I see change notes are published using GitHub Releases, however I could not find how the changes are recorded/maintained between releases. Here I propose to use a new `CHANGELOG.md` file which is tracked in the repository.

Or previously it was used Draft releases to track changes?

Comment (schollii) [reply RELEASING.rst:16]: @nicoddemus actually I don't recall setting that up, someone else might have. Whatever you find easier is fine by me, like whatever you are using for other projects. As long as we only have one place for the info, so there might be some info from previous releases to migrate?

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+name: deploy
+
+on:
+  push:
+    tags:
+      - "v*"
```
Comment (schollii) [inline .github/workflows/deploy.yml:6]: Can the pattern be a little more specific like a regex `^v([0-9]+\.){2}[0-9]+-pypi` so it's obvious the intent and one is much less likely to release to pypi by mistake.

**Review by schollii — APPROVED** [review]
Just one recommended improvement if possible so pre-approved

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (nicoddemus) [reply .github/workflows/deploy.yml:6]: In pytest we use `"[0-9]+.[0-9]+.[0-9]+"`, as it is more usual to use the version directly as tags -- having a `pypi` suffix in the tag is a bit unusual.

I suggest we use `"v[0-9]+.[0-9]+.[0-9]+"` here, which matches the current tags in the repository, and is unlikely to be a mistake. What do you think?

File: RELEASING.rst (hunk 1 above)
Comment (nicoddemus) [reply RELEASING.rst:16]: > Whatever you find easier is fine by me, like whatever you are using for other projects.

Now that we have one GitHub Release, GitHub will insist to show that as the latest release in the side bar, even if we push later tags:

image

So I suggest we keep using it, it is simple enough.

> As long as we only have one place for the info, so there might be some info from previous releases to migrate?

I did not find any unfortunately.

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (schollii) [reply .github/workflows/deploy.yml:6]: That's fine

**Comment by schollii** [comment]
Thanks @nicoddemus!

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (nicoddemus) [reply .github/workflows/deploy.yml:6]: I take you did not want to change to `"v[0-9]+.[0-9]+.[0-9]+"` in the end?

**Comment by nicoddemus** [comment]
Btw want to kick-off a new release? I'm just not sure what we would put in the changelog.

**Comment by schollii** [comment]
> Btw want to kick-off a new release? I'm just not sure what we would put in the changelog.

Sure. How about just say that there are no changes to the source code itself, just release process improvements.

Btw yes I thought the change about the branch pattern was there, so if you can please do it. Strange how some comments are just not visible in mobile chrome display of this PR. Like right now I can't see your other comment where you ask whether I still want pattern fixed. So I'm replying here.

**Comment by nicoddemus** [comment]
> How about just say that there are no changes to the source code itself, just release process improvements.

Actually the released version on [PyPI is `1.0.8` ](https://pypi.org/project/nose2pytest/), while the GitHub release page says `1.0.10`. :thinking: 

We might just release `1.0.11` saying that we are fixing the package upload to PyPI, what do you think?

**Pull request merged**